clib/cMap.ml

```diff
@@ -37,6 +37,10 @@ sig
   val height : 'a t -> int
   val filter_range : (key -> int) -> 'a t -> 'a t
   val of_list : (key * 'a) list -> 'a t
+  val symmetric_diff :
+    eq:('a -> 'a -> bool) ->
+    (key -> [ `Left of 'a | `Right of 'a | `Unequal of 'a * 'a ] -> 'b -> 'b) ->
```

Comment:
Don't use polymorphic variants, they're annoying as hell. You could define a common inductive in CSig or Utils, probably.

Reply:
Or better: just follow the Map.merge convention, which takes two 'a options. That would be even more regular.

Reply:
The problem with the `Map.merge` convention is that the API leaves the possibility for both `'a option`'s to be `None`, which can never happen. Are you sure you want to have that?

Reply:
I don't like Map.merge because of this, but uniformity is better than one bad API and one good API.

Reply:
Also polyvariants like this will have an extra layer of boxing compared to regular variants.